src/helpers.js

```diff
@@ -144,9 +144,27 @@ function registerWith(Handlebars) {
             );
         }
 
-        var intlData = options.data.intl || {},
-            locales  = intlData.locales,
-            formats  = intlData.formats;
+        var intlData      = options.data.intl || {},
+            locales       = intlData.locales,
+            formats       = intlData.formats,
+            hasSafeString = false;
+
+        var key, value;
+
+        for (key in hash) {
+            if (hash.hasOwnProperty(key)) {
+                value = hash[key];
+
+                if (!hasSafeString && value instanceof SafeString) {
+                    hasSafeString = true;
+                }
+
+                // Escape string value.
+                else if (typeof value === 'string') {
```

Comment:
I guess when the second `SafeString` is checked it'll be `typeof value === 'object'`.